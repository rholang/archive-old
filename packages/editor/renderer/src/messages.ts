import { defineMessages } from 'react-intl';

export const headingAnchorLinkMessages = defineMessages({
  copyHeadingLinkToClipboard: {
    id: 'fabric.editor.headingLink.copyAnchorLink',
    defaultMessage: 'Copy this anchor link',
    description: 'Copy heading link to clipboard',
  },
  copiedHeadingLinkToClipboard: {
    id: 'fabric.editor.headingLink.copied',
    defaultMessage: 'Copied!',
    description: 'Copied heading link to clipboard',
  },
  failedToCopyHeadingLink: {
    id: 'fabric.editor.headingLink.failedToCopy',
    defaultMessage: 'Copy failed',
    description: 'failed to copy heading link to clipboard',
  },
});
