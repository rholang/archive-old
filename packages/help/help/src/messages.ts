import { defineMessages } from 'react-intl';

export const messages = defineMessages({
  help_panel_header: {
    id: 'global_help.header',
    defaultMessage: 'Help',
    description: '',
  },
  help_panel_close: {
    id: 'global_help.close',
    defaultMessage: 'Close',
    description: '',
  },
  help_panel_navigation_back: {
    id: 'global_help.navigation.back',
    defaultMessage: 'Back',
    description: '',
  },
  help_panel_search_placeholder: {
    id: 'global_help.search.placeholder',
    defaultMessage: 'Search for help',
    description: '',
  },
  help_panel_article_rating_title: {
    id: 'global_help.article_rating.title',
    defaultMessage: 'Was this helpful?',
    description: '',
  },
  help_panel_article_rating_option_yes: {
    id: 'global_help.article_rating.option.yes',
    defaultMessage: 'Yes',
    description: '',
  },
  help_panel_article_rating_option_no: {
    id: 'global_help.article_rating.option.no',
    defaultMessage: 'No',
    description: '',
  },
  help_panel_article_rating_form_title: {
    id: 'global_help.article_rating.form.title',
    defaultMessage: 'Tell us why you gave this rating?',
    description: '',
  },
  help_panel_article_rating_accurate: {
    id: 'global_help.article_rating.accurate',
    defaultMessage: `It isn't accurate`,
    description: '',
  },
  help_panel_article_rating_clear: {
    id: 'global_help.article_rating.clear',
    defaultMessage: `It isn't clear`,
    description: '',
  },
  help_panel_article_rating_relevant: {
    id: 'global_help.article_rating.relevant',
    defaultMessage: `It isn't relevant`,
    description: '',
  },
  help_panel_article_rating_form_submit: {
    id: 'global_help.article_rating.form.submit',
    defaultMessage: `Submit`,
    description: '',
  },
  help_panel_article_rating_form_cancel: {
    id: 'global_help.article_rating.form.cancel',
    defaultMessage: `Cancel`,
    description: '',
  },

  help_panel_related_article_title: {
    id: 'global_help.related_article.title',
    defaultMessage: 'RELATED',
    description: '',
  },
  help_panel_related_article_show_more: {
    id: 'global_help.related_article.show_more',
    defaultMessage: 'Show more related',
    description: '',
  },
  help_panel_related_article_show_less: {
    id: 'global_help.related_article.show_less',
    defaultMessage: 'Show less related',
    description: '',
  },
  help_panel_search_results_title: {
    id: 'global_help.search_results.title',
    defaultMessage: 'Search results',
    description: '',
  },
  help_panel_search_results_suggestions: {
    id: 'global_help.search_results.suggestions',
    defaultMessage: 'SUGGESTIONS',
    description: '',
  },
  help_panel_search_results_no_results: {
    id: 'global_help.search_results.no_results',
    defaultMessage: `No matching search results`,
  },
  help_panel_search_error_loading_title: {
    id: 'global_help.error_loading.title',
    defaultMessage: `Something's gone wrong`,
    description: '',
  },
  help_panel_search_error_loading_text: {
    id: 'global_help.error_loading.text',
    defaultMessage: `Something prevented the page from loading. We will try to reconect automatically`,
    description: '',
  },
  help_panel_search_error_loading_button_label: {
    id: 'global_help.error_loading.button_label',
    defaultMessage: `Try now`,
    description: '',
  },
});
