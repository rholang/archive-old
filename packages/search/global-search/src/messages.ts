import { defineMessages } from 'react-intl';

export const messages = defineMessages({
  confluence_no_results_in_space: {
    id: 'global_search.confluence.no_results_in_space',
    defaultMessage: "No results in '{spaceTitle}'",
    description: '',
  },
  confluence_remove_space_filter: {
    id: 'global_search.confluence.remove_space_filter',
    defaultMessage: 'Remove space filter',
    description: '',
  },
  confluence_space_filter: {
    id: 'global_search.confluence.space_filter',
    defaultMessage: 'Filter results by space',
    description: '',
  },
  confluence_more_filters: {
    id: 'global_search.confluence.more_filters',
    defaultMessage: 'More filter options',
    description:
      'Text to display on button that takes the user to advanced search, which has more filtering options',
  },
  confluence_recent_pages_heading: {
    id: 'global_search.confluence.recent_pages_heading',
    defaultMessage: 'Recent pages and blogs',
    description: '',
  },
  confluence_recent_spaces_heading: {
    id: 'global_search.confluence.recent_spaces_heading',
    defaultMessage: 'Recent spaces',
    description: '',
  },
  confluence_search_placeholder: {
    id: 'global_search.confluence.search_placeholder',
    defaultMessage: 'Search Confluence',
    description: '',
  },
  confluence_spaces_heading: {
    id: 'global_search.confluence.spaces_heading',
    defaultMessage: 'Spaces',
    description: '',
  },
  confluence_confluence_objects_heading: {
    id: 'global_search.confluence.confluence_objects_heading',
    defaultMessage: 'Pages, blogs and attachments',
    description: '',
  },
  confluence_advanced_search: {
    id: 'global_search.confluence.advanced_search',
    defaultMessage: 'Advanced search',
    description: '',
  },
  confluence_advanced_search_filters: {
    id: 'global_search.confluence.advanced_search_filters',
    defaultMessage: 'Advanced search with filters',
    description: '',
  },
  confluence_advanced_search_for: {
    id: 'global_search.confluence.advanced_search_for',
    defaultMessage: 'Advanced search for "{query}"',
    description: '',
  },
  no_recent_activity_title: {
    id: 'global_search.no_recent_activity_title',
    defaultMessage: 'Search for what you need',
    description: '',
  },
  no_recent_activity_body: {
    id: 'global_search.no_recent_activity_body',
    defaultMessage:
      'Or use <a href={url}>advanced search</a> (`shift + enter`) to focus your results.',
    description: '',
  },
  no_results_title: {
    id: 'global_search.no_results_title',
    defaultMessage: "We couldn't find anything matching your search",
    description: '',
  },
  no_results_body: {
    id: 'global_search.no_results_body',
    defaultMessage:
      'Try again with a different term, or refine your results with our advanced search.',
    description: '',
  },
  people_recent_people_heading: {
    id: 'global_search.people.recent_people_heading',
    defaultMessage: 'Recently worked with',
    description: '',
  },
  people_people_heading: {
    id: 'global_search.people.people_heading',
    defaultMessage: 'People',
    description: '',
  },
  people_advanced_search: {
    id: 'global_search.people.advanced_search',
    defaultMessage: 'Search people',
    description: '',
  },
  search_error_title: {
    id: 'global_search.search_error_title',
    defaultMessage: "We're having trouble searching.",
    description: '',
  },
  search_error_body: {
    id: 'global_search.search_error_body',
    defaultMessage: 'It might just be hiccup. Best thing is to {link}.',
    description: '',
  },
  search_error_body_link: {
    id: 'global_search.search_error_body.link',
    defaultMessage: 'try again',
    description: '',
  },
  jira_search_placeholder: {
    id: 'global_search.jira.search_placeholder',
    defaultMessage: 'Search Jira',
    description: '',
  },
  jira_recent_issues_heading: {
    id: 'global_search.jira.recent_issues_heading',
    defaultMessage: 'Recent issues',
    description: '',
  },
  jira_recent_people_heading: {
    id: 'global_search.jira.recent_people_heading',
    defaultMessage: 'Recently worked with',
    description: '',
  },
  jira_recent_containers: {
    id: 'global_search.jira.recent_containers',
    defaultMessage: 'Recent boards, projects and filters',
    description: '',
  },
  jira_recent_core_containers: {
    id: 'global_search.jira.recent_core_containers',
    defaultMessage: 'Recent projects and filters',
    description: '',
  },

  jira_search_result_issues_heading: {
    id: 'global_search.jira.search_result_issues_heading',
    defaultMessage: 'Issues',
    description: 'Plural of issue',
  },
  jira_search_result_containers_heading: {
    id: 'global_search.jira.search_result_containers_heading',
    defaultMessage: 'Boards, projects and filters',
    description: '',
  },
  jira_search_result_core_containers_heading: {
    id: 'global_search.jira.search_result_core_containers_heading',
    defaultMessage: 'Projects and filters',
    description: '',
  },
  jira_search_result_people_heading: {
    id: 'global_search.jira.search_result_people_heading',
    defaultMessage: 'People',
    description: '',
  },
  jira_advanced_search: {
    id: 'global_search.jira.advanced_search',
    defaultMessage: 'View all:',
    description: '',
  },
  jira_advanced_search_issues: {
    id: 'global_search.jira.advanced_search_issues',
    defaultMessage: 'Issues',
    description: 'Plural of issue',
  },
  jira_advanced_search_projects: {
    id: 'global_search.jira.advanced_search_projects',
    defaultMessage: 'Projects',
    description: 'Plural of project',
  },
  jira_advanced_search_boards: {
    id: 'global_search.jira.advanced_search_boards',
    defaultMessage: 'Boards',
    description: 'Plural of board',
  },
  jira_advanced_search_filters: {
    id: 'global_search.jira.advanced_search_filters',
    defaultMessage: 'Filters',
    description: 'Plural of filter',
  },
  jira_advanced_search_people: {
    id: 'global_search.jira.advanced_search_people',
    defaultMessage: 'People',
    description: '',
  },
  jira_advanced_issue_search: {
    id: 'global_search.jira.advanced_issue_search',
    defaultMessage: 'Advanced search for issues',
    description: '',
  },
  jira_view_all_issues: {
    id: 'global_search.jira.view_all_issues',
    defaultMessage: 'View all issues',
    description: 'Advanced search for issues in post query screen',
  },
  jira_no_results_title: {
    id: 'global_search.jira.no_results_title',
    defaultMessage: "We couldn't find anything matching your search",
    description: '',
  },
  jira_no_results_body: {
    id: 'global_search.jira.no_results_body',
    defaultMessage: 'Try again with a different term.',
    description: '',
  },
  jira_no_recent_activity_body: {
    id: 'global_search.jira.no_recent_activity_body',
    defaultMessage: 'Or refine your results with our advanced search.',
    description: '',
  },
  jira_project_type_business_project: {
    id: 'global_search.jira.project_type.business_project',
    defaultMessage: 'Business Project',
    description: '',
  },
  jira_project_type_software_project: {
    id: 'global_search.jira.project_type.software_project',
    defaultMessage: 'Software Project',
    description: '',
  },
  jira_project_type_service_desk_project: {
    id: 'global_search.jira.project_type.service_desk_project',
    defaultMessage: 'Service Desk Project',
    description: '',
  },
  jira_project_type_ops_project: {
    id: 'global_search.jira.project_type.ops_project',
    defaultMessage: 'Ops Project',
    description: '',
  },
  jira_result_type_board: {
    id: 'global_search.jira.result_type.board',
    defaultMessage: 'Board',
    description: '',
  },
  jira_result_type_filter: {
    id: 'global_search.jira.result_type.filter',
    defaultMessage: 'Filter',
    description: '',
  },
  give_feedback: {
    id: 'global_search.give_feedback',
    defaultMessage: 'Give feedback',
    description: '',
  },
  show_more_button_text: {
    id: 'global_search.show_more_button_text',
    defaultMessage: 'Show {itemsPerPage} more results',
    description:
      'Text for button that is used to load more results for a search',
  },
  show_more_button_advanced_search: {
    id: 'global_search.show_more_button_advanced_search_v2',
    defaultMessage: 'View all results',
    description:
      'Text for button that is used when we cannot load more results and direct user to advanced search',
  },
  confluence_container_subtext_with_modified_date: {
    id: 'global_search.confluence.subtext_with_modified_date',
    defaultMessage: '{containerName}  Â·  Updated {friendlyLastModified}',
    description:
      'Text describing which space a given search result is from, as well as a date when it was last edited or created.',
  },
});
