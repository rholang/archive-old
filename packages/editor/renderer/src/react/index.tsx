import * as React from 'react';
// @ts-ignore: unused variable
// prettier-ignore
import { ComponentType, Consumer, Provider } from 'react';
import { Fragment, Mark, MarkType, Node, Schema } from 'prosemirror-model';
import { Serializer } from '../';
import { RendererAppearance } from '../ui/Renderer/types';
import { AnalyticsEventPayload } from '../analytics/events';

import {
  Doc,
  mergeTextNodes,
  isTextWrapper,
  TextWrapper,
  toReact,
} from './nodes';

import { toReact as markToReact } from './marks';
import {
  ProviderFactory,
  getMarksByOrder,
  isSameMark,
  EventHandlers,
  ExtensionHandlers,
  calcTableColumnWidths,
} from '@atlaskit/editor-common';
import { getText } from '../utils';

export interface RendererContext {
  objectAri?: string;
  containerAri?: string;
  adDoc?: any;
  schema?: Schema;
}

export interface ConstructorParams {
  providers?: ProviderFactory;
  eventHandlers?: EventHandlers;
  extensionHandlers?: ExtensionHandlers;
  portal?: HTMLElement;
  objectContext?: RendererContext;
  appearance?: RendererAppearance;
  disableHeadingIDs?: boolean;
  allowDynamicTextSizing?: boolean;
  allowHeadingAnchorLinks?: boolean;
  allowColumnSorting?: boolean;
  fireAnalyticsEvent?: (event: AnalyticsEventPayload) => void;
  shouldOpenMediaViewer?: boolean;
}

type MarkWithContent = Partial<Mark<any>> & {
  content: Array<MarkWithContent | Node<any>>;
};

function mergeMarks(marksAndNodes: Array<MarkWithContent | Node>) {
  return marksAndNodes.reduce(
    (acc, markOrNode) => {
      const prev = (acc.length && acc[acc.length - 1]) || null;

      if (
        markOrNode.type instanceof MarkType &&
        prev &&
        prev.type instanceof MarkType &&
        Array.isArray(prev.content) &&
        isSameMark(prev as Mark, markOrNode as Mark)
      ) {
        prev.content = mergeMarks(
          prev.content.concat((markOrNode as MarkWithContent).content),
        );
      } else {
        acc.push(markOrNode);
      }

      return acc;
    },
    [] as Array<MarkWithContent | Node>,
  );
}

export default class ReactSerializer implements Serializer<JSX.Element> {
  private providers?: ProviderFactory;
  private eventHandlers?: EventHandlers;
  private extensionHandlers?: ExtensionHandlers;
  private portal?: HTMLElement;
  private rendererContext?: RendererContext;
  private appearance?: RendererAppearance;
  private disableHeadingIDs?: boolean;
  private headingIds: string[] = [];
  private allowDynamicTextSizing?: boolean;
  private allowHeadingAnchorLinks?: boolean;
  private allowColumnSorting?: boolean;
  private fireAnalyticsEvent?: (event: AnalyticsEventPayload) => void;
  private shouldOpenMediaViewer?: boolean;

  constructor({
    providers,
    eventHandlers,
    extensionHandlers,
    portal,
    objectContext,
    appearance,
    disableHeadingIDs,
    allowDynamicTextSizing,
    allowHeadingAnchorLinks,
    allowColumnSorting,
    fireAnalyticsEvent,
    shouldOpenMediaViewer,
  }: ConstructorParams) {
    this.providers = providers;
    this.eventHandlers = eventHandlers;
    this.extensionHandlers = extensionHandlers;
    this.portal = portal;
    this.rendererContext = objectContext;
    this.appearance = appearance;
    this.disableHeadingIDs = disableHeadingIDs;
    this.allowDynamicTextSizing = allowDynamicTextSizing;
    this.allowHeadingAnchorLinks = allowHeadingAnchorLinks;
    this.allowColumnSorting = allowColumnSorting;
    this.fireAnalyticsEvent = fireAnalyticsEvent;
    this.shouldOpenMediaViewer = shouldOpenMediaViewer;
  }

  private resetState() {
    this.headingIds = [];
  }

  serializeFragment(
    fragment: Fragment,
    props: any = {},
    target: any = Doc,
    key: string = 'root-0',
    parentInfo?: { parentIsIncompleteTask: boolean; path: Array<Node> },
  ): JSX.Element | null {
    // This makes sure that we reset internal state on re-render.
    if (key === 'root-0') {
      this.resetState();
    }

    const content = ReactSerializer.getChildNodes(fragment).map(
      (node, index) => {
        if (isTextWrapper(node)) {
          return this.serializeTextWrapper(node.content);
        }

        let props;

        if (node.type.name === 'table') {
          props = this.getTableProps(node);
        } else if (node.type.name === 'date') {
          props = this.getDateProps(node, parentInfo);
        } else if (node.type.name === 'heading') {
          props = this.getHeadingProps(node, parentInfo && parentInfo.path);
        } else if (['tableHeader', 'tableRow'].indexOf(node.type.name) > -1) {
          props = this.getTableChildrenProps(node);
        } else if (node.type.name === 'media') {
          props = this.getMediaProps(node);
        } else {
          props = this.getProps(node);
        }

        let currentPath = (parentInfo && parentInfo.path) || [];

        const parentIsIncompleteTask =
          node.type.name === 'taskItem' && node.attrs.state !== 'DONE';

        let pInfo = {
          parentIsIncompleteTask,
          path: [...currentPath, node],
        };

        const serializedContent = this.serializeFragment(
          node.content,
          props,
          toReact(node),
          `${node.type.name}-${index}`,
          pInfo,
        );

        if (node.marks && node.marks.length) {
          return ([] as Array<Mark>)
            .concat(node.marks)
            .reverse()
            .reduce((acc, mark) => {
              return this.renderMark(
                markToReact(mark),
                this.getMarkProps(mark),
                `${mark.type.name}-${index}`,
                acc,
              );
            }, serializedContent);
        }

        return serializedContent;
      },
    );

    return this.renderNode(target, props, key, content);
  }

  private serializeTextWrapper(content: Node[]) {
    return ReactSerializer.buildMarkStructure(content).map((mark, index) =>
      this.serializeMark(mark, index),
    );
  }

  private serializeMark(mark: Mark, index: number = 0) {
    if (mark.type.name === 'text') {
      return (mark as any).text;
    }

    const content = ((mark as any).content || []).map(
      (child: Mark, index: number) => this.serializeMark(child, index),
    );
    return this.renderMark(
      markToReact(mark),
      this.getMarkProps(mark),
      `${mark.type.name}-${index}`,
      content,
    );
  }

  private renderNode(
    NodeComponent: ComponentType<any>,
    props: any,
    key: string,
    content: string | JSX.Element | any[] | null | undefined,
  ): JSX.Element {
    return (
      <NodeComponent key={key} {...props}>
        {content}
      </NodeComponent>
    );
  }

  private renderMark(
    MarkComponent: ComponentType<any>,
    props: any,
    key: string,
    content: any,
  ) {
    return (
      <MarkComponent key={key} {...props}>
        {content}
      </MarkComponent>
    );
  }

  private getTableChildrenProps(node: Node) {
    return {
      ...this.getProps(node),
      allowColumnSorting: this.allowColumnSorting,
    };
  }

  private getTableProps(node: Node) {
    return {
      ...this.getProps(node),
      allowColumnSorting: this.allowColumnSorting,
      columnWidths: calcTableColumnWidths(node),
      tableNode: node,
    };
  }

  private getDateProps(
    node: Node,
    parentInfo: { parentIsIncompleteTask: boolean } | undefined,
  ) {
    return {
      timestamp: node.attrs && node.attrs.timestamp,
      parentIsIncompleteTask: parentInfo && parentInfo.parentIsIncompleteTask,
    };
  }
  private getMediaProps(node: Node) {
    return {
      ...this.getProps(node),
      shouldOpenMediaViewer: this.shouldOpenMediaViewer,
    };
  }

  private getProps(node: Node) {
    return {
      text: node.text,
      providers: this.providers,
      eventHandlers: this.eventHandlers,
      extensionHandlers: this.extensionHandlers,
      portal: this.portal,
      rendererContext: this.rendererContext,
      serializer: this,
      content: node.content ? node.content.toJSON() : undefined,
      allowDynamicTextSizing: this.allowDynamicTextSizing,
      allowHeadingAnchorLinks: this.allowHeadingAnchorLinks,
      rendererAppearance: this.appearance,
      fireAnalyticsEvent: this.fireAnalyticsEvent,
      ...node.attrs,
    };
  }

  private headingAnchorSupported(path: Array<Node> = []): boolean {
    return (
      path.length === 0 || path[path.length - 1].type.name === 'layoutColumn'
    );
  }

  private getHeadingProps(node: Node, path: Array<Node> = []) {
    return {
      ...node.attrs,
      content: node.content ? node.content.toJSON() : undefined,
      headingId: this.getHeadingId(node),
      showAnchorLink:
        this.allowHeadingAnchorLinks &&
        !this.disableHeadingIDs &&
        this.headingAnchorSupported(path),
    };
  }

  // The return value of this function is NOT url encoded,
  // In HTML5 standard, id can contain any characters, encoding is no necessary.
  // Plus we trying to avoid double encoding, therefore we leave the value as is.
  // Remember to use encodeURIComponent when generating url from the id value.
  private getHeadingId(node: Node) {
    if (this.disableHeadingIDs || !node.content.size) {
      return;
    }

    // We are not use node.textContent here, because we would like to handle cases where
    // headings only contain inline blocks like emoji, status and date.
    const nodeContent = (node as any).content
      .toJSON()
      .reduce((acc: string, node: any) => acc.concat(getText(node) || ''), '')
      .trim()
      .replace(/\s/g, '-');

    if (!nodeContent) {
      return;
    }

    return this.getUniqueHeadingId(nodeContent);
  }

  private getUniqueHeadingId(baseId: string, counter = 0): string {
    if (counter === 0 && this.headingIds.indexOf(baseId) === -1) {
      this.headingIds.push(baseId);
      return baseId;
    } else if (counter !== 0) {
      const headingId = `${baseId}.${counter}`;
      if (this.headingIds.indexOf(headingId) === -1) {
        this.headingIds.push(headingId);
        return headingId;
      }
    }

    return this.getUniqueHeadingId(baseId, ++counter);
  }

  private getMarkProps(mark: Mark): any {
    const { key, ...otherAttrs } = mark.attrs;
    return {
      eventHandlers: this.eventHandlers,
      markKey: key,
      ...otherAttrs,
    };
  }

  static getChildNodes(fragment: Fragment): (Node | TextWrapper)[] {
    const children: Node[] = [];
    fragment.forEach(node => {
      children.push(node);
    });
    return mergeTextNodes(children) as Node[];
  }

  static getMarks(node: Node): Mark[] {
    if (!node.marks || node.marks.length === 0) {
      return [];
    }

    return getMarksByOrder(node.marks);
  }

  static buildMarkStructure(content: Node[]) {
    return mergeMarks(
      content.map(node => {
        const nodeMarks = this.getMarks(node);
        if (nodeMarks.length === 0) {
          return node;
        }

        return nodeMarks.reverse().reduce(
          (acc, mark) => {
            const { eq } = mark;

            return {
              ...mark,
              eq,
              content: [acc],
            };
          },
          node as any,
        );
      }),
    ) as Mark[];
  }

  static fromSchema(
    _schema: Schema,
    {
      providers,
      eventHandlers,
      extensionHandlers,
      appearance,
      disableHeadingIDs,
      allowDynamicTextSizing,
      allowHeadingAnchorLinks,
      allowColumnSorting,
    }: ConstructorParams,
  ): ReactSerializer {
    // TODO: Do we actually need the schema here?
    return new ReactSerializer({
      providers,
      eventHandlers,
      extensionHandlers,
      appearance,
      disableHeadingIDs,
      allowDynamicTextSizing,
      allowHeadingAnchorLinks,
      allowColumnSorting,
    });
  }
}
