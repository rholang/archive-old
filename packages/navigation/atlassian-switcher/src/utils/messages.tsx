import * as React from 'react';
import { defineMessages } from 'react-intl';
import FormattedMessage from '../primitives/formatted-message';

import IntlProvider from '../components/intl-provider';

const messages = defineMessages({
  switchTo: {
    id: 'fabric.atlassianSwitcher.switchTo',
    defaultMessage: 'Switch to',
    description:
      'In a context in which users are able to switch between products, this text is the title of the category displaying the products the user is able to switch to.',
  },
  switchToTooltip: {
    id: 'fabric.atlassianSwitcher.switchToTooltip',
    defaultMessage: 'Switch to…',
    description:
      'This text appears as a tooltip when a user hovers over the atlassian switcher icon before clicking on it.',
  },
  recent: {
    id: 'fabric.atlassianSwitcher.recent',
    defaultMessage: 'Recent',
    description:
      "In a context in which users are able to view recent projects or spaces they've viewed, this text is the title of the section displaying all the recent projects or spaces.",
  },
  more: {
    id: 'fabric.atlassianSwitcher.more',
    defaultMessage: 'More',
    description:
      'In a context in which users are able to view predefined custom links, this text is the title of the section displaying all existing custom links.',
  },
  try: {
    id: 'fabric.atlassianSwitcher.try',
    defaultMessage: 'Try',
    description:
      'This text appears as a way to encourage the user to try a new Atlassian product.',
  },
  manageList: {
    id: 'fabric.atlassianSwitcher.manageList',
    defaultMessage: 'Manage list',
    description:
      'This text is for the action for a user to manage the values present in an editable list of links.',
  },
  jiraProject: {
    id: 'fabric.atlassianSwitcher.jiraProject',
    defaultMessage: 'Jira project',
    description:
      'In a context in which several items are listed , this text describes that the specific type of a given item is a Jira project',
  },
  confluenceSpace: {
    id: 'fabric.atlassianSwitcher.confluenceSpace',
    defaultMessage: 'Confluence space',
    description:
      'In a context in which several items are listed , this text describes that the specific type of a given item is a Confluence space',
  },
  administration: {
    id: 'fabric.atlassianSwitcher.administration',
    defaultMessage: 'Administration',
    description:
      'The text of a link redirecting the user to the site administration',
  },
  discoverMore: {
    id: 'fabric.atlassianSwitcher.discoverMore',
    defaultMessage: 'More Atlassian products',
    description:
      'The text of a link redirecting the user to Discover More Atlassian products',
  },
  browseApps: {
    id: 'fabric.atlassianSwitcher.browseApps',
    defaultMessage: 'Browse Marketplace apps',
    description:
      'The text of a link redirecting the user to Discover Embedded Marketplace within in the product (Marketplace is a brand name. Please dont translate it)',
  },
  errorHeading: {
    id: 'fabric.atlassianSwitcher.errorHeading',
    defaultMessage: 'Something’s gone wrong',
    description:
      'Heading of the error screen which is shown when an unknown error happens in the Atlassian Switcher. Usually due to failed network requests.',
  },
  errorText: {
    id: 'fabric.atlassianSwitcher.errorText',
    defaultMessage:
      'We keep track of these errors, but feel free to contact us if refreshing doesn’t fix things',
    description:
      'Text that is displayed when an unknown error happens in the Atlassian Switcher.',
  },
  errorImageAltText: {
    id: 'fabric.atlassianSwitcher.errorImageAltText',
    defaultMessage: 'A broken robot and a number of people busy fixing it.',
    description:
      'Text displayed as alt text when an error occurs in the Atlassian Switcher',
  },
  errorTextNetwork: {
    id: 'fabric.atlassianSwitcher.errorTextNetwork',
    defaultMessage:
      'We couldn’t load this list. Please reload the page and try again.',
    description: 'Text that is displayed when we detect a network issue.',
  },
  errorTextLoggedOut: {
    id: 'fabric.atlassianSwitcher.errorTextLoggedOut',
    defaultMessage: 'You’ve been logged out. Please log in again.',
    description: 'Text that is displayed when we detect user is logged out.',
  },
  login: {
    id: 'fabric.atlassianSwitcher.login',
    defaultMessage: 'Log in',
    description: 'Text in log in button.',
  },
  showMoreSites: {
    id: 'fabric.atlassianSwitcher.show.more.sites',
    defaultMessage: 'Show more sites',
    description: 'The text of a toggle showing more site options',
  },
  discover: {
    id: 'fabric.atlassianSwitcher.discover',
    defaultMessage: 'Discover',
    description: 'The header of "Discover" section',
  },
  productDescriptionConfluence: {
    id: 'fabric.atlassianSwitcher.product.description.confluence',
    defaultMessage: 'Document collaboration',
    description: 'Text displayed under Confluence product recommendation.',
  },
  productDescriptionJiraServiceDesk: {
    id: 'fabric.atlassianSwitcher.product.description.jsd',
    defaultMessage: 'IT service desk and customer service',
    description:
      'Text displayed under Jira Service Desk product recommendation.',
  },
  productDescriptionJiraSoftware: {
    id: 'fabric.atlassianSwitcher.product.description.jsw',
    defaultMessage: 'Project and issue tracking',
    description: 'Text displayed under Jira Software product recommendation.',
  },
  productDescriptionOpsgenie: {
    id: 'fabric.atlassianSwitcher.product.description.opsgenie',
    defaultMessage: 'Modern incident management',
    description: 'Text displayed under Opsgenie product recommendation.',
  },
});

export const SwitchToTooltipText = (
  <IntlProvider>
    <FormattedMessage {...messages.switchToTooltip} />
  </IntlProvider>
);

export default messages;
