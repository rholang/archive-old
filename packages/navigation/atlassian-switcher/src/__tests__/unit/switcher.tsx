import * as React from 'react';
import { shallow } from 'enzyme';
import Switcher from '../../components/switcher';
import messages from '../../utils/messages';

const noop = () => void 0;

describe('Switcher', () => {
  it('should render sections with headers by default', () => {
    expect(
      shallow(
        <Switcher
          messages={messages}
          triggerXFlow={noop}
          hasLoaded
          hasLoadedCritical
          onDiscoverMoreClicked={noop}
          licensedProductLinks={[]}
          suggestedProductLinks={[]}
          fixedLinks={[]}
          adminLinks={[]}
          recentLinks={[]}
          customLinks={[]}
          discoverSectionLinks={[]}
        />,
      ),
    ).toMatchSnapshot();
  });

  it('should be able to disable headers with "disableHeadings"', () => {
    expect(
      shallow(
        <Switcher
          messages={messages}
          triggerXFlow={noop}
          hasLoaded
          hasLoadedCritical
          onDiscoverMoreClicked={noop}
          licensedProductLinks={[]}
          suggestedProductLinks={[]}
          fixedLinks={[]}
          adminLinks={[]}
          recentLinks={[]}
          customLinks={[]}
          disableHeadings
          discoverSectionLinks={[]}
        />,
      ),
    ).toMatchSnapshot();
  });
});
