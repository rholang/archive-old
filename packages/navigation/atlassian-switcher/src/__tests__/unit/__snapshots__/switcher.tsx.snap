// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Switcher should be able to disable headers with "disableHeadings" 1`] = `
<NavigationAnalyticsContext
  data={
    Object {
      "attributes": Object {
        "itemsCount": 0,
      },
    }
  }
>
  <SwitcherWrapper
    appearance="drawer"
  >
    <WithAnalyticsEvents(class_2)
      data={
        Object {
          "adminLinks": Array [],
          "fixedLinks": Array [],
          "licensedProducts": Array [],
          "numberOfSites": 0,
          "suggestedProducts": Array [],
        }
      }
      subject="atlassianSwitcher"
    />
    <Component
      sectionId="switchTo"
      title={null}
    />
    )}
    <Component
      sectionId="recent"
      title={null}
    />
    <Component
      sectionId="customLinks"
      title={null}
    />
  </SwitcherWrapper>
</NavigationAnalyticsContext>
`;

exports[`Switcher should render sections with headers by default 1`] = `
<NavigationAnalyticsContext
  data={
    Object {
      "attributes": Object {
        "itemsCount": 0,
      },
    }
  }
>
  <SwitcherWrapper
    appearance="drawer"
  >
    <WithAnalyticsEvents(class_2)
      data={
        Object {
          "adminLinks": Array [],
          "fixedLinks": Array [],
          "licensedProducts": Array [],
          "numberOfSites": 0,
          "suggestedProducts": Array [],
        }
      }
      subject="atlassianSwitcher"
    />
    <Component
      sectionId="switchTo"
      title={
        <Unknown
          defaultMessage="Switch to"
          description="In a context in which users are able to switch between products, this text is the title of the category displaying the products the user is able to switch to."
          id="fabric.atlassianSwitcher.switchTo"
        />
      }
    />
    )}
    <Component
      sectionId="recent"
      title={
        <Unknown
          defaultMessage="Recent"
          description="In a context in which users are able to view recent projects or spaces they've viewed, this text is the title of the section displaying all the recent projects or spaces."
          id="fabric.atlassianSwitcher.recent"
        />
      }
    />
    <Component
      sectionId="customLinks"
      title={
        <Unknown
          defaultMessage="More"
          description="In a context in which users are able to view predefined custom links, this text is the title of the section displaying all existing custom links."
          id="fabric.atlassianSwitcher.more"
        />
      }
    />
  </SwitcherWrapper>
</NavigationAnalyticsContext>
`;
